FirebaseMessaging/Sources/Token/FIRMessagingTokenManager.m

```diff
@@ -102,23 +102,28 @@ - (NSString *)defaultFCMToken {
 
 - (void)setDefaultFCMToken:(NSString *)defaultFcmToken {
   // Should always trigger the token refresh notification when the delegate method is called
+  // No need to check if the token has changed, it's handled in the notification receiver.
   NSNotificationCenter *center = [NSNotificationCenter defaultCenter];
   [center postNotificationName:kFIRMessagingRegistrationTokenRefreshNotification
                         object:defaultFcmToken];
 }
 
 - (void)saveDefaultTokenInfo:(NSString *)defaultFcmToken {
-  _defaultFCMToken = [defaultFcmToken copy];
-  FIRMessagingTokenInfo *tokenInfo =
-      [[FIRMessagingTokenInfo alloc] initWithAuthorizedEntity:_fcmSenderID
-                                                        scope:kFIRMessagingDefaultTokenScope
-                                                        token:defaultFcmToken
-                                                   appVersion:FIRMessagingCurrentAppVersion()
-                                                firebaseAppID:_firebaseAppID];
-  tokenInfo.APNSInfo =
-      [[FIRMessagingAPNSInfo alloc] initWithTokenOptionsDictionary:[self tokenOptions]];
-
-  [self->_tokenStore saveTokenInfoInCache:tokenInfo];
+  if (_defaultFCMToken.length != defaultFcmToken.length ||
+      (_defaultFCMToken.length && defaultFcmToken.length &&
+       ![_defaultFCMToken isEqualToString:defaultFcmToken])) {
```

Comment:
nit: I see this non-trivial check again. Would it make sense to move it to a helper to improve readability and reduce code duplication?

Reply:
Done.